function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautAwsProxyRequest.java

```diff
@@ -193,6 +194,14 @@ public URI getUri() {
         return URI.create(getScheme() + "://" + hostHeader + path);
     }
 
+    @Nonnull
+    @Override
+    public InetSocketAddress getRemoteAddress() {
+        final String sourceIp = this.awsProxyRequest.getRequestContext().getIdentity().getSourceIp();
```

Comment:
Can you add null checks here because it seems there is a possibility that `getRequestContext()` and/or `getIdentity()` could be `null`

If the source IP cannot be resolved you can fall back to the super implementation